### PR TITLE
Uses a replicated copy of HTTP solver

### PR DESCRIPTION
**Pull request by crdant**
TL;DR
-----

Configures the workload cluster to use a replicated copy of the
cert manager HTTP solver

Details
-------

I ran into an issue where certificates were not being issued in
my workload cluster because of the image policy. The pod that 
responds to the challenge from Let's Encrypt was unable to run
in namespaces that were restricted to only run workloads from
the private registry.

Two solutions presented themselves, switch to a DNS challenge or
replicating the solver image into our private registry. The DNS
challenge is my usually approach, but there's a simple elegance
to the HTTP solver that I wanted to maintain. I found out that a
command-line flag for the cert manager controller allowed me to 
change the image, so I decided to use that option plus a local
replica of the image to solve the problem.

This change includes the Harbor setup for the replication and 
a script `patch-cert-manager.sh` that patches cert manager on
the workload cluster to use it.

### CODE REVIEW

**Comment by crdant** [comment]
TL;DR
-----

Configures the workload cluster to use a replicated copy of the
cert manager HTTP solver

Details
-------

I ran into an issue where certificates were not being issued in
my workload cluster because of the image policy. The pod that 
responds to the challenge from Let's Encrypt was unable to run
in namepsaces that were restricted to only run workloads from
the private registry.

Two solutions presented themselves, switch to a DNS challenge or
replicating the solver image into our private registry. The DNS
challenge is my usually approach, but there's a simple elegance
to the HTTP solver that I wanted to maintain. I found out that a
command-line flag for the cert manager controller allowed me to 
change the image, so I decided to use that option plus a local
replicat of the image to solve the problem.

This change includes the Harbor setup for the replication and 
a script `patch-cert-manager.sh` that patches cert manager on
the workload cluster to use it.
